### PR TITLE
Backend JPA and testing 

### PR DESCRIPTION
**Pull request by shravanvemula**

### CODE REVIEW

File: creating-joy/pom.xml [hunk 1]
```diff
@@ -68,13 +80,33 @@
...
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<version>1.4.194</version>
+		</dependency>
+		<dependency>
+			<groupId>org.hibernate</groupId>
+			<artifactId>hibernate-entitymanager</artifactId>
+			<version>5.2.9.Final</version>
+		</dependency>
+		<dependency>
```
Comment (ghost) [inline creating-joy/pom.xml:90]: Junit is already part of spring

File: creating-joy/pom.xml [hunk 2]
```diff
@@ -68,13 +80,33 @@
...
+		</dependency>
 		<dependency>
-			<groupId>org.springframework.security</groupId>
-			<artifactId>spring-security-test</artifactId>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<version>1.4.194</version>
+		</dependency>
+		<dependency>
+			<groupId>org.hibernate</groupId>
```
Comment (ghost) [inline creating-joy/pom.xml]: entitiy manager is already part of spring

**Review by ghost — CHANGES_REQUESTED** [review]

File: creating-joy/pom.xml (hunk 1 above)
Comment (shravanvemula) [reply creating-joy/pom.xml:90]: Removed junit dependency

File: creating-joy/pom.xml (hunk 2 above)
Comment (shravanvemula) [reply creating-joy/pom.xml]: removed entity manager dependency

File: creating-joy/src/test/java/com/technocrats/creatingjoy/repository/UserRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@Slf4j
+public class UserRepositoryTest {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private AddressRepository addressRepository;
```
Comment (venkatdevara09) [inline creating-joy/src/test/java/com/technocrats/creatingjoy/repository/UserRepositoryTest.java]: Hi,
As Atif has suggested, do not use other repositories other than the repository you are testing.

Comment (shravanvemula) [reply creating-joy/src/test/java/com/technocrats/creatingjoy/repository/UserRepositoryTest.java]: I've tested whether one to one mapping is working or not for address and user

**Review by venkatdevara09 — APPROVED** [review]

File: creating-joy/src/test/java/com/technocrats/creatingjoy/repository/UserRepositoryTest.java (hunk 3 above)
Comment (venkatdevara09) [reply creating-joy/src/test/java/com/technocrats/creatingjoy/repository/UserRepositoryTest.java]: Please remove addressRepository and fetch address using user object itself

**Review by venkatdevara09 — APPROVED** [review]

File: creating-joy/src/test/java/com/technocrats/creatingjoy/repository/UserRepositoryTest.java (hunk 3 above)
Comment (shravanvemula) [reply creating-joy/src/test/java/com/technocrats/creatingjoy/repository/UserRepositoryTest.java]: Done

**Review by Hemasree5c2 — APPROVED** [review]

**Pull request merged**